cmake/CMakeLists.txt

```diff
@@ -92,6 +92,20 @@ if(COLVARS_OPENMP)
   endif()
 endif()
 
+find_package(Python3)
+if(Python3_FOUND)
+  execute_process(
+    COMMAND ${Python3_EXECUTABLE} -c "import sys; print(next(p for p in sys.path if 'site-packages' in p))"
+    OUTPUT_VARIABLE SITE_PACKAGES_PATH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  find_package(Torch PATHS file(JOIN ${SITE_PACKAGES_PATH} "torch/share/cmake/Torch"))
+  if(Torch_FOUND)
+    message(STATUS "Will link against Torch: ${TORCH_LIBRARY}")
+    target_include_directories(colvars PRIVATE ${TORCH_INCLUDE_PATH})
+    target_link_libraries(colvars PRIVATE ${TORCH_LIBRARY})
+  endif()
+endif()
```

Comment:
How those lines interact with the other search for Torch below in the CmakeLists.txt? (see lines 140-152)

Reply:
I tried to build covlars as a library: 

- Torch_FOUND is true, libcolvars.a can be created, but torchann class is not compiled, because -DTORCH is not set. 

-  If I change the value of COLVARS_TORCH (line 138) to ON, then torchann class will be compiled and libcolvars.a can be created. However,  there are several errors "undefined reference to..." in the step "Linking CXX executable run_colvars_test".

Reply:
I integrated the two, but I am also finding linkage errors that shouldn't be there (undefined symbol when running `ld` while also including `libcolvars.so`, which does define that symbol).

Unless somebody has other ideas, I would keep an eye out for what the MD packages will do in their own test suites. We will have to use their build systems for distribution anyway...

Setting this back as draft.

Reply:
looks like it's not very straightforward :/

According to #570, it looks like only the libTorch (C++) is needed. Does the Python version is mandatory? Otherwise it could simplify the cmake recipe